In [1]:
from pathlib import Path
from abstract_search_evaluation import get_scores

/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
MODELS_PATH = Path("models")

In [10]:
def print_model_scores(model_path: str | Path):
    scores = get_scores(model_path)
    print(model_path)
    print(f"Full match score: {scores["perfect_match_score"] * 100:.2f}%")
    print(f"In top results score: {scores["in_top_results_score"] * 100:.2f}%")

In [11]:
for model_path in MODELS_PATH.iterdir():
    print_model_scores(model_path)
    print("")

You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


Loading weights: 100%|██████████| 199/199 [00:00<00:00, 876.37it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/vanilla-finetuned
Full match score: 6.03%
In top results score: 27.57%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 1007.32it/s, Materializing param=pooler.dense.weight]                              
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-0.50
Full match score: 5.83%
In top results score: 28.56%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 1029.93it/s, Materializing param=pooler.dense.weight]                              
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-0.75
Full match score: 6.13%
In top results score: 28.95%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 971.59it/s, Materializing param=pooler.dense.weight]                               
You are trying to use a model that was created with Sentence Transformers version 5.3.0, but you're currently using version 5.2.3. This might cause unexpected behavior or errors. In that case, try to update to the latest version.


models/jargon-layman-finetuned-1.00
Full match score: 6.23%
In top results score: 30.14%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 1001.20it/s, Materializing param=pooler.dense.weight]                              


models/jargon-layman-finetuned-0.25
Full match score: 6.23%
In top results score: 28.75%



In [12]:
print_model_scores("allenai/scibert_scivocab_uncased")

No sentence-transformers model found with name allenai/scibert_scivocab_uncased. Creating a new one with mean pooling.
Loading weights: 100%|██████████| 199/199 [00:00<00:00, 1144.54it/s, Materializing param=pooler.dense.weight]                               
BertModel LOAD REPORT from: allenai/scibert_scivocab_uncased
Key                                        | Status     |  | 
-------------------------------------------+------------+--+-
cls.predictions.transform.dense.bias       | UNEXPECTED |  | 
cls.predictions.transform.dense.weight     | UNEXPECTED |  | 
cls.predictions.decoder.bias               | UNEXPECTED |  | 
cls.predictions.decoder.weight             | UNEXPECTED |  | 
cls.seq_relationship.bias                  | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.bias   | UNEXPECTED |  | 
cls.seq_relationship.weight                | UNEXPECTED |  | 
cls.predictions.bias                       | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.weight | UNEXPECTED |  

allenai/scibert_scivocab_uncased
Full match score: 0.89%
In top results score: 4.45%


# Results

| Model | Exact abstract match score | Abstract in top 5 results score |
|---|---|---|
| allenai/scibert_scivocab_uncased | 0.89% | 4.45% |
| vanilla-finetuned | **6.03%** | **27.57%** |
| jargon-layman-finetuned | <u>6.23%</u> | <u>30.14%</u> |